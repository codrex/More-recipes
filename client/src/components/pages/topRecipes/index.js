import TopRecipes from './topRecipes';

export default TopRecipes;
