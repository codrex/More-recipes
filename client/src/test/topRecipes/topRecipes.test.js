import React from 'react';
import toJson from 'enzyme-to-json';
import { PureTopRecipes } from '../../components/pages/topRecipes/topRecipes';

const props = {
  getTopRecipes: jest.fn(),
  topRecipes: [{
    name: 'capachino',
    category: 'drinks',
    views: 1,
    upVotes: 1,
    downVotes: 1,
    id: 1,
    image: {},
  }],
  user: {
    id: 1,
    favRecipes: [{
      id: 1,
    }],
  },
};

describe('TopRecipes page component ', () => {
  test('render as expected ', () => {
    const wrapper = shallow(<PureTopRecipes {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
});
