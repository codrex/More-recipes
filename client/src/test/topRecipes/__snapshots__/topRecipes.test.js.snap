// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopRecipes page component  render as expected  1`] = `
<Connect(RecipesDisplay)
  getRecipes={[Function]}
  getTopRecipes={[Function]}
  recipes={
    Array [
      Object {
        "category": "drinks",
        "downVotes": 1,
        "id": 1,
        "image": Object {},
        "name": "capachino",
        "upVotes": 1,
        "views": 1,
      },
    ]
  }
  title="Top recipes"
  topRecipes={
    Array [
      Object {
        "category": "drinks",
        "downVotes": 1,
        "id": 1,
        "image": Object {},
        "name": "capachino",
        "upVotes": 1,
        "views": 1,
      },
    ]
  }
  user={
    Object {
      "favRecipes": Array [
        Object {
          "id": 1,
        },
      ],
      "id": 1,
    }
  }
/>
`;
